cumulusci/tasks/bulkdata/load.py

```diff
@@ -320,14 +325,13 @@ def _generate_results_id_map(self, step, local_ids):
             if result.success:
                 yield (local_id, result.id)
             else:
-                if self.options["ignore_row_errors"]:
-                    self.logger.warning(
-                        f"Error on record with id {local_id}: {result.error}"
-                    )
-                else:
-                    raise BulkDataException(
-                        f"Error on record with id {local_id}: {result.error}"
-                    )
+                check_for_row_error(
+                    self,
```

Comment:
I find the way this function stashes state inside the task unintuitive. Lacking that, the function doesn't need a reference to the task at all. Could we just use a local counter?

Reply:
(Otherwise, I'm fine with this change).

Reply:
Okay I got rid of the weirdness but it did push some extra complexity into the Task class. I wasn't totally happy the way it was but not the way it is either. I probably should have just left it as two files that shared 10 lines of identical code. Which was also not ideal.